src/structify/resources/structure.py

```diff
@@ -259,6 +259,23 @@ def run_async(
             ),
             cast_to=object,
         )
+    
+    def run(
+        self,
+        *args,
+        **kwargs,
+    ) -> object:
+        # This is a manually added function. Its goal is to simulate a synchronous run of the async function.
+        # by calling it and then waiting
+        token = self.run_async(*args, **kwargs)
+
+        while True:
```

Comment:
Maybe take in a timeout here, and if we hit that timeout either try to cancel the job or just return?

Reply:
Yup. Adding.